src/commands.rs

```diff
@@ -210,22 +250,34 @@ impl <'a> CommandInner<'a> for AnalyzeCommand {
 
     // ANALYZE - execution
     fn run(&self, client: &ApiClient, args: AnalyzeCommandArgs<'a>) -> CommandResult {
-        let AnalyzeCommandArgs { project_id, files, name } = args;
-
-        // no matter what, start the analysis
-        let mut analysis_response: ApiResult<ApiAnalysisJobResponse> = client
-            .start_analysis(project_id, files)
-            .map(|resp| {
-                println!("# Started analysis {} with job id {}", resp.analysis_id, resp.job_id);
-                resp
-            });
+        let AnalyzeCommandArgs { project_context, branch_name, include_git_source, git_branch_name, files, name } = args;
+
+        let mut analysis_response: ApiResult<ApiAnalysisJobResponse> =
+            if include_git_source || git_branch_name.is_some() {
+                client.start_analysis_with_git(project_context.clone(), branch_name, include_git_source, git_branch_name, files)
+                    .and_then(|resp| {
+                        println!("Requesting new analysis with job id {} with included git source", resp.job_id);
+                        client.poll_job_completion(&resp.job_id, Duration::from_secs(2))?;
+                        client.get_job_result(&resp.job_id).map(|result| {
+                            println!("# Started analysis {} with job id {} with included git source", result.analysis_id, resp.job_id);
```

Comment:
Seems like the `#` at the beginning of the printlns is a convention. Been a while since I initially wrote this stuff but it's probably to make it easier for end users to filter out "debug output" lines.